sszgen/generator/hash.go

```diff
@@ -138,7 +138,7 @@ func (v *Value) hashTreeRoot(name string, appendBytes bool) string {
 			})
 		} else {
 			// dynamic bytes require special handling, need length mixed in
-			hMethod := "PutBytes"
+			hMethod := "Append"
```

Comment:
The [hash tree root](https://github.com/ferranbt/fastssz/blob/main/spectests/structs_encoding.go#L6136) of `PreviousEpochParticipation` and  `CurrentEpochParticipation` does not work. Both `PutBytes` and `MerkleizeWithMixin` merklizes the bytes if the bytes length is > 32, resulting in a double merklization. The beacon spec tests doesn't uncover this bug because all the bytes in the spec tests for these 2 values are less than 32 bytes.

Reply:
Can this happen in production? It might be because `Prysm` does not use `fastssz` for the `BeaconState` and this case is not covered.

Reply:
Yes, it can happen in production. I believe that is correct. `HashTreeRoot` for the `BeaconState` crashed before it could get to this part of the code (it crashed at index 11, and this happens on index 15), so I think that's why we haven't encountered it before.